opts/envdir.go

```diff
@@ -0,0 +1,56 @@
+package opts
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+/*
+Read in a directory with files acting as environment variables.
+*/
+func ParseEnvDir(directory string) ([]string, error) {
+	dir, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return []string{}, err
+	}
+
+	lines := []string{}
+	for _, info := range dir {
+		variable := info.Name()
+		// ignore directories and hidden files
+		if !info.IsDir() && variable[0] != '.' {
+			if strings.ContainsAny(variable, whiteSpaces) {
+				return []string{}, ErrBadEnvVariable{fmt.Sprintf("variable '%s' has white spaces", variable)}
+			}
+
+			data, err := getFirstLine(filepath.Join(directory, variable))
+			if err != nil {
+				return []string{}, err
+			}
+
+			lines = append(lines, fmt.Sprintf(
+				"%s=%s",
+				variable,
+				// null bytes are to be converted to newlines according to envdir
+				bytes.TrimSpace(bytes.Replace(data, []byte{0}, []byte{'\n'}, -1)),
```

Comment:
I'd prefer if you did this clean-up logic in `getFirstLine` just to keep this call to append() a little less complex.